### PR TITLE
feat(surveys): Improved survey creation flow

### PR DESCRIPTION
**Pull request by smallbrownbike**
https://github.com/PostHog/posthog/assets/28248250/626fa363-acfe-4e54-b148-76c55fde4ef7

### CODE REVIEW

File: frontend/src/scenes/surveys/Survey.tsx [hunk 1]
```diff
@@ -100,447 +77,7 @@ export function SurveyForm({ id }: { id: string }): JSX.Element {
...
-                                    appearance={value || defaultSurveyAppearance}
-                                />
-                            )}
-                        </Field>
-                    ) : (
-                        <SurveyAPIEditor survey={survey} />
-                    )}
-                </div>
-            </div>
+            {true ? <EditSurveyNew /> : <EditSurveyOld />}
```
Comment (smallbrownbike) [inline frontend/src/scenes/surveys/Survey.tsx:80]: @liyiy Feature flag can go here!

**Comment by liyiy** [comment]
@smallbrownbike Would you like some help with merge conflicts and the feature flagging here?

**Comment by liyiy** [comment]
Other than fixing failing checks and adding the feature flag, I gave the collapse panel header a background color so it's easier to parse and labeled the question indexes. The rest remains the same. Awesome work from @smallbrownbike !

Screen Shot 2023-10-13 at 2 54 34 PM

**Review by liyiy — APPROVED** [review]
Well I also fixed this up so would like another approval from someone else 😅 ... but for the stuff that was mainly @smallbrownbike 's work, 👍

**Comment by neilkakkar** [comment]
all targeting wasn't working well because of some pre-existing bugs that made this worse (the linked flag wasn't being removed) and when you go to edit an existing survey it removed all targeting.

Buut, these should be resolved now, also updated cypress tests, seems to work fine, let's get this in and quickly switch over so we don't have to maintain old & new code for long 👍

File: frontend/src/scenes/surveys/EditSurveyNew.tsx [hunk 2]
```diff
@@ -0,0 +1,969 @@
...
+                                                                                                        onChange(
+                                                                                                            newChoices
+                                                                                                        )
+                                                                                                    }}
+                                                                                                />
+                                                                                            </div>
+                                                                                        )
+                                                                                    )}
+                                                                                    <div className="w-fit">
+                                                                                        {(value || []).length < 6 && (
```
Comment (marandaneto) [inline frontend/src/scenes/surveys/EditSurveyNew.tsx:549]: Curious why 6 here, do you know @neilkakkar @jurajmajerik or @smallbrownbike ?
Later this was relaxed for APIs at least https://github.com/PostHog/posthog/pull/26224/

**Pull request merged**